tests/components/template/test_lock.py

```diff
@@ -283,65 +170,171 @@ def test_lock_action(self):
+                    },
                 }
             },
         )
 
-        self.hass.block_till_done()
-        self.hass.start()
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
 
-        self.hass.states.set("switch.test_state", STATE_ON)
-        self.hass.block_till_done()
+    assert hass.states.async_all() == []
 
-        state = self.hass.states.get("lock.template_lock")
-        assert state.state == lock.STATE_LOCKED
 
-        self.hass.services.call(
-            lock.DOMAIN, lock.SERVICE_UNLOCK, {ATTR_ENTITY_ID: "lock.template_lock"}
+async def test_template_static(hass, caplog):
+    """Test that we allow static templates."""
+    with assert_setup_component(1, lock.DOMAIN):
+        assert await setup.async_setup_component(
+            hass,
+            lock.DOMAIN,
+            {
+                "lock": {
+                    "platform": "template",
+                    "value_template": "{{ 1 + 1 }}",
+                    "lock": {
+                        "service": "switch.turn_on",
+                        "entity_id": "switch.test_state",
+                    },
+                    "unlock": {
+                        "service": "switch.turn_off",
+                        "entity_id": "switch.test_state",
+                    },
+                }
+            },
         )
-        self.hass.block_till_done()
 
-        assert len(self.calls) == 1
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("lock.template_lock")
+    assert state.state == lock.STATE_UNLOCKED
+
+    hass.states.async_set("lock.template_lock", lock.STATE_LOCKED)
+    await hass.async_block_till_done()
+    state = hass.states.get("lock.template_lock")
+    assert state.state == lock.STATE_LOCKED
+
+
+async def test_lock_action(hass, calls):
+    """Test lock action."""
+    assert await setup.async_setup_component(
+        hass,
+        lock.DOMAIN,
+        {
+            "lock": {
+                "platform": "template",
+                "value_template": "{{ states.switch.test_state.state }}",
+                "lock": {"service": "test.automation"},
+                "unlock": {
+                    "service": "switch.turn_off",
+                    "entity_id": "switch.test_state",
+                },
+            }
+        },
+    )
+
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    hass.states.async_set("switch.test_state", STATE_OFF)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("lock.template_lock")
+    assert state.state == lock.STATE_UNLOCKED
+
+    await hass.services.async_call(
+        lock.DOMAIN, lock.SERVICE_LOCK, {ATTR_ENTITY_ID: "lock.template_lock"}
+    )
+    await hass.async_block_till_done()
+
+    assert len(calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)